src/item_contents.cpp

```diff
@@ -1018,6 +1018,83 @@ std::vector<const item *> item_contents::gunmods() const
     return mods;
 }
 
+std::vector<const item *> item_contents::mods() const
+{
+    std::vector<const item *> mods;
+    for( const item_pocket &pocket : contents ) {
+        if( pocket.is_type( item_pocket::pocket_type::MOD ) ) {
+            for( const item *it : pocket.all_items_top() ) {
+                mods.insert( mods.end(), it );
+            }
+        }
+    }
+    return mods;
+}
+
+void item_contents::update_modified_pockets(
+    const cata::optional<const pocket_data *> &mag_or_mag_well,
+    std::vector<const pocket_data *> container_pockets )
+{
+    for( auto pocket_iter = contents.begin(); pocket_iter != contents.end(); ) {
+        item_pocket &pocket = *pocket_iter;
+        if( pocket.is_type( item_pocket::pocket_type::CONTAINER ) ) {
+
+            const pocket_data *current = pocket.get_pocket_data();
+            bool found = false;
+            // this loop is to make sure the pockets on the current item are already here from @container_pockets,
+            // so we don't need to clear them (saving the favorite data)
+            for( auto container_pocket = container_pockets.begin(); container_pocket != container_pockets.end();
```

Comment:
This could probably using some standard algorithm
```suggestion
           const auto iter = std::find( container_pockets.begin(), container_pockets.end(), current );
           if( iter != container_pockets.end()) { erase(iter); } else { ... }
```

Reply:
@KorGgenT has merged my branch into his changes for #41917 so I will be closing my PR. @BevapDin would you mind making this suggestion on his PR instead?